changelog.d/13801.feature

```diff
@@ -0,0 +1 @@
+Add `listeners[x].request_id_header` config to specify which request header to extract and use as the request ID in order to correlate requests from a reverse-proxy.
```

Comment:
Looks good!

I'm not sure it is necessary, but it could be worth also cross linking from https://github.com/matrix-org/synapse/blob/918c74bfb57e3ca4d300ed9a3bfb99b99126f821/docs/usage/administration/request_log.md?plain=1#L19

Up to you whether you feel that would be useful!

Reply:
I'm going to pass on cross-linking this. I can't think of a way to add that makes things better and more clear than before. The information about "(This identifier is shared by related log lines)" is still accurate with or without this change.

The cross-link between the reverse-proxy docs is great though!

Reply:
Actually, linking from the config documentation to the request log format would be good though. Added ✅

I still can't think of a way to add extra info to the request log format page though.

Reply:
That's fine. Let's ship it!

Reply:
:shipit: Just waiting for the CI